src/wavm/wavm.cc

```diff
@@ -90,16 +91,35 @@ namespace {
       WAVM::Runtime::catchRuntimeExceptions(                                                       \
           [&] { _x; },                                                                             \
           [&](WAVM::Runtime::Exception *exception) {                                               \
-            auto description = describeException(exception);                                       \
-            _wavm->fail(FailState::RuntimeError,                                                   \
-                        "Function: " + std::string(function_name) + " failed: " + description);    \
-            destroyException(exception);                                                           \
+            _wavm->fail(FailState::RuntimeError, getFailMessage(function_name, exception));        \
             throw std::exception();                                                                \
           });                                                                                      \
     } catch (...) {                                                                                \
     }                                                                                              \
   } while (0)
 
+std::string getFailMessage(std::string_view function_name, WAVM::Runtime::Exception *exception) {
+  std::string message = "Function " + std::string(function_name) + " failed:\n" +
+                        "WAVM message: " + WAVM::Runtime::describeExceptionType(exception->type) +
```

Comment:
Could you pull the message into the first line and remove `WAVM message:`?

Also, maybe switch the `wasm backtrace:` to `Proxy-Wasm plugin in-VM backtrace:` to make it clear that it covers only the plugin part and not host.

The example message should look like this:
```
[2020-10-28 14:24:10.551][979699][trace][wasm] [source/extensions/common/wasm/wasm_vm.cc:29] Function proxy_on_tick failed: wavm.reachedUnreachable
Proxy-Wasm plugin in-VM backtrace:
  1: wasm!helloworld!runtime._panic+7
  2: wasm!helloworld!command-line-arguments.C+2
  3: wasm!helloworld!command-line-arguments.B+0
  4: wasm!helloworld!command-line-arguments.A+0
  5: wasm!helloworld!(*command-line-arguments.helloWorld).OnTick+74
  6: wasm!helloworld!proxy_on_tick+0
```

Reply:
fixed 
![Screenshot from 2020-10-29 18-36-57](https://user-images.githubusercontent.com/13513977/97551049-d11e6e80-1a15-11eb-8e89-a43c36faec70.png)